sale_service_project/models/sale.py

```diff
@@ -93,25 +111,48 @@ def _compute_task_closed(self):
             line.task_closed = all(line.mapped('task_ids.stage_id.closed'))
 
     @api.multi
+    @api.depends('task_work_ids', 'task_work_product_id')
+    def _compute_task_work_values(self):
+        for line in self.filtered(
+                lambda x: x.task_work_ids and x.task_work_product_id):
+            pricelist = line.order_id.pricelist_id
+            partner = line.order_id.partner_id
+            line.task_work_hours = sum(line.task_work_ids.mapped('hours'))
+            line.task_work_product_price = pricelist.price_get(
+                line.task_work_product_id.id, line.task_work_hours,
+                partner.id)[pricelist.id]
+            line.task_work_amount = (
+                line.task_work_hours * line.task_work_product_price)
+
+    @api.multi
+    @api.depends('task_materials_ids')
+    def _compute_task_materials_amount(self):
+        for line in self.filtered(lambda x: x.task_materials_ids):
+            materials_amount = 0.0
+            for material in line.task_materials_ids:
+                materials_amount += material.quantity * material.price
+            line.task_materials_amount = materials_amount
+
+    @api.multi
     def product_id_change(
-        self, pricelist, product, qty=0, uom=False, qty_uos=0, uos=False,
+        self, pricelist, product_id, qty=0, uom=False, qty_uos=0, uos=False,
```

Comment:
Don't rename this. Upstream variables are wrong, I know, but there can be people that uses all arguments with name (indeed, I remember one in Odoo), and renaming this you'll get an error.